Cognite.Simulator.Utils/ConfigurationLibraryBase.cs

```diff
@@ -175,88 +182,88 @@ protected virtual T StateFromRoutineRevision(SimulatorRoutineRevision routineRev
-                    SampleExternalId = its.SaveTimeseriesExternalId
-                }),
-                OutputTimeSeries = routineRev.Configuration.OutputTimeseries.Select(ots => new OutputTimeSeriesConfiguration()
-                {
-                    Name = ots.Name,
-                    Type = ots.ReferenceId,
-                    Unit = ots.Unit,
-                    UnitType = ots.UnitType,
-                    ExternalId = ots.SaveTimeseriesExternalId
-                }),
-                DataSampling = new DataSamplingConfiguration()
-                {
-                    ValidationWindow = routineRev.Configuration.DataSampling.ValidationWindow,
-                    SamplingWindow = routineRev.Configuration.DataSampling.SamplingWindow,
-                    Granularity = routineRev.Configuration.DataSampling.Granularity,
-                    ValidationEndOffset = routineRev.Configuration.DataSampling.ValidationEndOffset
-                },
-                LogicalCheck = new LogicalCheckConfiguration()
-                {
-                    Enabled = routineRev.Configuration.LogicalCheck.Enabled,
-                    ExternalId = routineRev.Configuration.LogicalCheck.TimeseriesExternalId,
-                    AggregateType = routineRev.Configuration.LogicalCheck.Aggregate,
-                    Check = routineRev.Configuration.LogicalCheck.Operator,
-                    Value = routineRev.Configuration.LogicalCheck.Value ?? 0 // TODO what's the default value here?
-                },
-                SteadyStateDetection = new SteadyStateDetectionConfiguration()
-                {
-                    Enabled = routineRev.Configuration.SteadyStateDetection.Enabled,
-                    ExternalId = routineRev.Configuration.SteadyStateDetection.TimeseriesExternalId,
-                    AggregateType = routineRev.Configuration.SteadyStateDetection.Aggregate,
-                    MinSectionSize = routineRev.Configuration.SteadyStateDetection.MinSectionSize ?? 0, // TODO what's the default value here?
-                    VarThreshold = routineRev.Configuration.SteadyStateDetection.VarThreshold ?? 0, // TODO what's the default value here?
-                    SlopeThreshold = routineRev.Configuration.SteadyStateDetection.SlopeThreshold ?? 0 // TODO what's the default value here?
-                },
-                UserEmail = "",
-                Routine = routineRev.Script.Select((s, i) => new CalculationProcedure()
-                {
-                    Order = i,
-                    Steps = s.Steps.Select((step, j) => new CalculationProcedureStep()
-                    {
-                        Step = j,
-                        Type = step.StepType,
-                        Arguments = step.Arguments
-                    })
-                }),
-                CreatedTime = routineRev.CreatedTime
-            };
-            localConfiguration = LocalConfigurationFromRoutine(simulationConfigurationWithRoutine);
```

Comment:
the biggest change is here basically. we don't use `SimulationConfigurationWithRoutine` anymore